controllers/array_action/array_mediator_svc.py

```diff
@@ -1353,14 +1493,32 @@ def _delete_rcrelationship(self, rcrelationship_id):
             else:
                 logger.warning("failed to delete rcrelationship '{0}': {1}".format(rcrelationship_id, ex))
 
-    def delete_replication(self, replication_name):
+    def delete_replication(self, replication):
+        if replication.replication_type == array_settings.REPLICATION_TYPE_MIRROR:
+            self._delete_replication(replication.name)
+        elif replication.replication_type == array_settings.REPLICATION_TYPE_EAR:
+            self._delete_ear_replication(replication.volume_group_id)
+
+    def _delete_replication(self, replication_name):
         rcrelationship = self._get_rcrelationship_by_name(replication_name, not_exist_error=False)
         if not rcrelationship:
             logger.info("could not find replication with name {}".format(replication_name))
             return
         self._stop_rcrelationship(rcrelationship.id)
         self._delete_rcrelationship(rcrelationship.id)
 
+    def _delete_ear_replication(self, volume_group_id):
+        if not self._is_earreplication_supported():
+            logger.info("EAR replication is not supported on the existing storage")
+            return
+
+        self._change_volume_group_policy(volume_group_id)
+
+        # for phase 1 - move volume outside the group and delete the volume group
```

Comment:
same

Reply:
I removed comment and added explanation to design document in wiki